unexpected status 400 with error: ValidationError: One or more fields contain incorrect values

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.0.2

### AzureRM Provider Version

3.71.0

### Affected Resource(s)/Data Source(s)

"azurerm_api_management_api" "apis"

### Terraform Configuration Files

```hcl
Error: creating/updating Api (Subscription: "xxxxxxxxxxxxxxxxxxxx"
│ Resource Group Name: "rgrp-xxx-as-xxxgw"
│ Service Name: "apim-dva2-as-xxxx003"
│ Api: "index-quality-rrm-actuator-api-v1-0"): performing CreateOrUpdate: unexpected status 400 with error: ValidationError: One or more fields contain incorrect values:
│ 
│   with module.api.azurerm_api_management_api.apis[0],
│   on .terraform/modules/api/api-template/api-resources.tf line 3, in resource "azurerm_api_management_api" "apis":
│    3: resource "azurerm_api_management_api" "apis" {
```


### Debug Output/Panic Output

```shell
Executing terraform apply
total 64
drwxr-xr-x 3 vsts docker  4096 Aug 30 09:27 ..
-rwxrwxrwx 1 vsts docker   570 Aug 30 09:27 auth.tf
-rwxrwxrwx 1 vsts docker  2042 Aug 30 09:27 variables.tf
-rwxrwxrwx 1 vsts docker    23 Aug 30 09:27 outputs.tf
-rwxrwxrwx 1 vsts docker   629 Aug 30 09:27 api-import-specs.json
drwxrwxrwx 2 vsts docker  4096 Aug 30 09:27 v1.0
-rwxrwxrwx 1 vsts docker   441 Aug 30 09:27 main.tf
-rwxrwxrwx 1 vsts docker  1579 Aug 30 09:27 arguments-values.auto.tfvars
-rwxrwxrwx 1 vsts docker  1188 Aug 30 09:27 .terraform.lock.hcl
drwxrwxrwx 4 vsts docker  4096 Aug 30 09:27 .terraform
-rwxrwxrwx 1 vsts docker 14592 Aug 30 09:27 index-quality_rrm-actuator-api_terraform.plan
-rwxrwxrwx 1 vsts docker  1330 Aug 30 09:27 deployment-config.json
drwxrwxrwx 4 vsts docker  4096 Aug 30 09:27 .
module.api.azurerm_api_management_api.apis[0]: Modifying... [id=/subscriptions/xxxxxxxxxxxxxxx/resourceGroups/rgxas-xxxxxapigw/providers/Microsoft.ApiManagement/service/apim-xxxxxxx-apigw003/apis/xxxxxxxxxxxxxxxxxxxxxxx]
╷
│ Error: creating/updating Api (Subscription: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
│ Resource Group Name: "rxxxxxxxxxxxigw"
│ Service Name: "axxxxxxxxxxxxxxxxxxxxx03"
│ Api: "ixxxxxxxxxxxxxxxxxxxxxxxxx1-0"): performing CreateOrUpdate: unexpected status 400 with error: ValidationError: One or more fields contain incorrect values:
│ 
│   with module.api.azurerm_api_management_api.apis[0],
│   on .terraform/modules/api/api-template/api-resources.tf line 3, in resource "azurerm_api_management_api" "apis":
│    3: resource "azurerm_api_management_api" "apis" {
│ 
╵
##[debug]Exit code 1 received from tool '/usr/bin/bash'
##[debug]STDIO streams have closed for tool '/usr/bin/bash'
##[error]Bash exited with code '1'.
##[debug]Processed: ##vso[task.issue type=error;]Bash exited with code '1'.
##[debug]task result: Failed
##[debug]Processed: ##vso[task.complete result=Failed;done=true;]
```


### Expected Behaviour

The plan drawn by terraform should be applied on APIM.

### Actual Behaviour

Api: "ixxxxxxxxxxxxxxxxxxxxxxxxx1-0"): performing CreateOrUpdate: unexpected status 400 with error: ValidationError: One or more fields contain incorrect values:

### Steps to Reproduce

terraform init 
terraform plan

### Important Factoids

_No response_

### References

it is an intermittent issue.

I'm having the same issue on 3.70.0. Here is the ERROR log (not sure how to produce a 'normal' log):

>2023-09-12T13:50:29.809-0700 [ERROR] provider.terraform-provider-azurerm_v3.70.0_x5: Response contains error diagnostic: @module=sdk.proto diagnostic_detail= diagnostic_severity=ERROR diagnostic_summary="creating/updating Api (Subscription: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
Resource Group Name: "rg-ncus-openai-poc"
Service Name: "apim-ncus-openai-poc"
Api: "api-ncus-openai-poc"): performing CreateOrUpdate: unexpected status 400 with error: ValidationError: One or more fields contain incorrect values:" tf_rpc=ApplyResourceChange tf_resource_type=azurerm_api_management_api @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/diag/diagnostics.go:55 tf_proto_version=5.3 tf_provider_addr=provider tf_req_id=62af9bb0-a4c3-9553-320d-7fe2f9e4f9b2 timestamp=2023-09-12T13:50:29.809-0700
2023-09-12T13:50:29.816-0700 [ERROR] vertex "azurerm_api_management_api.openai[0]" error: creating/updating Api (Subscription: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
Resource Group Name: "rg-ncus-openai-poc"
Service Name: "apim-ncus-openai-poc"
Api: "api-ncus-openai-poc"): performing CreateOrUpdate: unexpected status 400 with error: ValidationError: One or more fields contain incorrect values:

--- a/website/docs/r/api_management_api.html.markdown
+++ b/website/docs/r/api_management_api.html.markdown
@@ -36,11 +36,6 @@ resource "azurerm_api_management_api" "example" {
   display_name        = "Example API"
   path                = "example"
   protocols           = ["https"]
-
-  import {
-    content_format = "swagger-link-json"
-    content_value  = "http://conferenceapi.azurewebsites.net/?format=json"
-  }
 }
 ```
 

--- a/website/docs/r/api_management_api.html.markdown
+++ b/website/docs/r/api_management_api.html.markdown
@@ -36,6 +36,10 @@ resource "azurerm_api_management_api" "example" {
   display_name        = "Example API"
   path                = "example"
   protocols           = ["https"]
+  import {
+    content_format = "swagger-link-json"
+    content_value  = "https://raw.githubusercontent.com/hashicorp/terraform-provider-azurerm/refs/heads/main/internal/services/apimanagement/testdata/api_management_api_schema_swagger.json"
+  }
 }
 ```
 

--- a/website/docs/r/api_management_api.html.markdown
+++ b/website/docs/r/api_management_api.html.markdown
@@ -111,7 +111,7 @@ A `import` block supports the following:
 
 * `content_format` - (Required) The format of the content from which the API Definition should be imported. Possible values are: `openapi`, `openapi+json`, `openapi+json-link`, `openapi-link`, `swagger-json`, `swagger-link-json`, `wadl-link-json`, `wadl-xml`, `wsdl` and `wsdl-link`.
 
-* `content_value` - (Required) The Content from which the API Definition should be imported. When a `content_format` of `*-link-*` is specified this must be a URL, otherwise this must be defined inline.
+* `content_value` - (Required) The Content from which the API Definition should be imported. When a `content_format` of `swagger-json` is specified this must be an accessible URL and returns a valid Swagger JSON document. When a `content_format` of `*-link-*` is specified this must be a URL, otherwise this must be defined inline. 
 
 * `wsdl_selector` - (Optional) A `wsdl_selector` block as defined below, which allows you to limit the import of a WSDL to only a subset of the document. This can only be specified when `content_format` is `wsdl` or `wsdl-link`.
 

--- a/website/docs/r/api_management_api.html.markdown
+++ b/website/docs/r/api_management_api.html.markdown
@@ -111,7 +111,7 @@ A `import` block supports the following:
 
 * `content_format` - (Required) The format of the content from which the API Definition should be imported. Possible values are: `openapi`, `openapi+json`, `openapi+json-link`, `openapi-link`, `swagger-json`, `swagger-link-json`, `wadl-link-json`, `wadl-xml`, `wsdl` and `wsdl-link`.
 
-* `content_value` - (Required) The Content from which the API Definition should be imported. When a `content_format` of `swagger-json` is specified this must be an accessible URL and returns a valid Swagger JSON document. When a `content_format` of `*-link-*` is specified this must be a URL, otherwise this must be defined inline. 
+* `content_value` - (Required) The Content from which the API Definition should be imported. When a `content_format` of `*-link-*` is specified this must be a URL, otherwise this must be defined inline. The URL must be accessible and return a valid document; otherwise, deployment may fail.
 
 * `wsdl_selector` - (Optional) A `wsdl_selector` block as defined below, which allows you to limit the import of a WSDL to only a subset of the document. This can only be specified when `content_format` is `wsdl` or `wsdl-link`.
 

